### PR TITLE
Ticket7406

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
### Description of work

When hovering over a matplotlib graph, coordinates of the hovered-over point are shown.

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/7406

### Acceptance criteria

See ticket

### Unit tests

Yes

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] If the change is to an OPI, does the `check_opi_format.py` script in [C:\Instrument\Dev\ibex_gui\base\uk.ac.stfc.isis.ibex.opis](https://github.com/ISISComputingGroup/ibex_gui/blob/master/base/uk.ac.stfc.isis.ibex.opis/check_opi_format.py) pass?
- [ ] Do the changes function as described and is it robust?
- [ ] Is there associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?

### Final Steps
- [ ] Reviewer has also merged the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibFigure.java [hunk 1]
```diff
@@ -66,8 +71,12 @@ public MatplotlibFigure(Composite parent, int style, String url, int figureNumbe
 		labelConnectionStatus.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
 		labelConnectionStatus.setText(viewModel.getPlotName().getValue());
 		
+		plotMessage = new Label(container, SWT.NONE);
+		plotMessage.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false));
+		plotMessage.setText("");
+		
 		plotCanvas = new Canvas(container, SWT.NONE);
-		plotCanvas.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		plotCanvas.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 2, 1));
```
Comment (ConorMFinn) [inline base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibFigure.java:79]: It's not clear to me what these numbers are doing and look a bit magic. Is this something that could be a named constant?

**Review by ConorMFinn — CHANGES_REQUESTED** [review]
Just a couple of minor code readability and usability things

File: base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibFigureViewModel.java [hunk 2]
```diff
@@ -131,20 +153,33 @@ private void updatePlotName() {
...
+	
+	/**
+	 * Updates the plot message from the model.
+	 */
+	public void updatePlotMessage() {
+		if (model.isConnected()) {
+			final var message = model.getPlotMessage();
+			plotMessage.setValue(message);
+		} else {
+			plotMessage.setValue("");
```
[the remark this answers is not in the document]
Comment (Tom-Willemsen) [reply base/uk.ac.stfc.isis.ibex.ui.graphing/src/uk/ac/stfc/isis/ibex/ui/graphing/websocketview/MatplotlibFigureViewModel.java:172]: > Would it be worth adding in a "Coordinates: " or similar here and above?

I don't think we want to do this - the "message" from matplotlib can actually be anything, e.g. "file saved successfully", there might even be a user API in matplotlib to set the message! The current implementation is analogous to what you would see in the native `webagg` backend (tip: while plots in ibex are open, if you go to 127.0.0.1:8988, you can see the MPL webagg implementation).

The goal is eventually for IBEX's implementation to be fully equivalent to WebAgg, we're not quite there yet (still need to add the toolbar buttons) but getting closer. But I don't think we want to deviate too much from what WebAgg does.

> It wasn't obvious this feature was even available until you waited with the cursor over the plot for a while

I've fixed this by using a slightly different eclipse API which reports every mouse move as opposed to "hover" events - it is now much more responsive which I think aids discoverability - and feels about as responsive as WebAgg.

**Review by ConorMFinn — APPROVED** [review]
Looks good to me, thanks for the info on the Webagg stuff 👍

**Pull request merged**